src/app/page/character/character.component.ts

```diff
@@ -0,0 +1,55 @@
+import { Component, OnInit, Input, OnChanges } from '@angular/core';
+import { CharacterService } from 'src/app/service/character.service';
+import { Character } from 'src/app/shared/model/character';
+import { forkJoin } from 'rxjs';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-character',
+  templateUrl: './character.component.html',
+  styleUrls: ['./character.component.css']
+})
+export class CharacterComponent implements OnInit, OnChanges {
+  @Input() listCharaterUrls: string[] ;
+  characters: Character[]
+  searchText;
+  isLoading: boolean = false
+  constructor(
+    private characterService: CharacterService,
+    private router: Router
+  ) { }
+
+  ngOnInit(): void {
+    this.getCharacters();
+  }
+
+  ngOnChanges():void{
+    this.getCharacters();
+  }
+
+  getCharacters(){
+    this.isLoading = true
+    if(this.listCharaterUrls !== null){
+      let listQuery = this.listCharaterUrls.map(url => this.characterService.getCharacter(url));
+      forkJoin(listQuery).subscribe(results => {
+           this.characters = results
+           this.isLoading = false
+          }
+        );
+    }
+    else{
+    this.characterService.getCharacters().subscribe(characters => {
+      this.characters = characters
+      this.isLoading = false
+    })
+    }
+   
+  }
+
+  selectCharacter(data){
+    localStorage.setItem('characterCurrent',data.url);
+    var id = data.url.split("/").slice(-1).pop()
+    this.router.navigate(['character-detail',id])
+  }
```

Comment:
this.router.navigate(['character-detail', data.url.split("/").slice(-1).pop()]);